views/map.js

```diff
@@ -147,6 +153,55 @@ class MapView extends Component {
 					'circle-color': VEHICLE_POINT_COLOR,
 				},
 			})
+
+			let popup = null
+			const showPopup = (e) => {
+				const feature = e.features[0]
+				if (!feature) return
+				const props = feature.properties
+
+				// Only show defined properties
+				const addProp = (label, value, code = true) => {
+					if (!value || value === 'null' || value === 'undefined') return ''
+					const displayValue = code ? `<code>${value}</code>` : value
+					return `${label}: ${displayValue}<br/>`
+				}
+
+				let vehicleSection = ''
+				vehicleSection += addProp('ID', props.vehicleId)
+				vehicleSection += addProp('Label', props.vehicleLabel)
+				vehicleSection += addProp('License Plate', props.vehicleLicensePlate)
+
+				let tripSection = ''
+				tripSection += addProp('ID', props.trip_id)
+				tripSection += addProp('Route', props.route_id)
+
+				const startDate = props.start_date.match(/^(\d{4})(\d{2})(\d{2})$/)
+				if (startDate && props.start_time) {
+					const start = new Date(Date.parse(`${startDate[1]}-${startDate[2]}-${startDate[3]}T${props.start_time}`))
+					// Default locale with short date/time
+					tripSection += addProp('Start', new Intl.DateTimeFormat(undefined, {
+						dateStyle: 'short',
+						timeStyle: 'short',
+					}).format(start), false)
```

Comment:
Could reuse [`renderStartDate` and `renderStartTime`](https://github.com/public-transport/gtfs-rt-inspector/blob/16ef1fdb07daa40a2ad20b44d630cd88ff06eedc/views/inspector.js#L6-L13), but this has nice locale support

Reply:
I think it would be best to have both the human-readable and the raw date/time, e.g. one as the regular text and the other as a tooltip (using `<abbr>`).